service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java

```diff
@@ -1021,6 +1035,26 @@ public ResponseEntity<ApiGcpDataprocClusterResource> updateDataprocCluster(
       }
     }
 
+    // Ensure that the cluster is in a "RUNNING" state
+    try {
```

Comment:
can you do this as part of the flight?

Reply:
Yeah I originally was gonna add this as a flight step but moved it here to keep all the validation checks together. I don't have a strong opinion so I can change if you want

Reply:
You recommend creating a new step for this. 
reason - makes code idempotent if its a step (see validation steps in AWS), handles retries and rollback if necessary.

Reply:
That makes a lot of sense. done.